### PR TITLE
XEP-280: Improve readability

### PR DESCRIPTION
**Pull request by penguineer**
In light of [CVE-2017-5589+ Multiple XMPP Clients User Impersonation Vulnerability](https://rt-solutions.de/en/2017/02/CVE-2017-5589_xmpp_carbons/) and after a chat with @ge0rg I came to the conclusion that similar mishaps can be avoided by improving readability of the requirements.

This PR is a proposal for restructuring the XEP so that each single requirement can be easily discerned from other requirements, i.e. each structural element does not contain more than one requirement (so you can check them each and cross them off the list), while still keeping overall readability as a document.

There are no changes to the actual content. I only moved some things around within a section, e.g. single short paragraphs after a screen full of examples, which could be easily overlooked. Here and there I changed some minor details for style reasons.

Please consider this PR as my contribution for helping others to better understand the intentions and boundaries of the XEP and to avoid security mishaps due to misunderstandings or omissions while implementing the XEP.

### CODE REVIEW

**Comment by ge0rg** [comment]
Thanks for your work, @penguineer. I like the changes very much and I'll be glad to redo #412 once this is accepted.

**Comment by linuxwolf** [comment]
I'm +1

:shipit:

File: xep-0280.xml [hunk 1]
```diff
@@ -372,37 +393,50 @@
...
+      <li>In order to prevent unwanted termination of conversations on other resources, clients SHOULD NOT send &lt;gone/&gt; chat states on logout, instead</li>
+      <li>clients SHOULD count on the broadcast of unavailable presence to convey the change in attention.</li>
       <li>Upon receiving an outbound notification of any chat state other than &lt;gone/&gt;, the copied client MAY conclude that the sending client has taken responsibility for the conversation, and make appropriate user interface modifications.  For example, notifications could be suppressed on devices receiving the Carbon copies.</li>
     </ul>
   </section2>
   <section2 topic='Handling of Errors' anchor='bizrules-errors'>
-    <p>When a receiving server attempts to deliver a forked message, and that message bounces with an error for any reason, the receiving server MUST NOT forward that error back to the original sender.  The receiving server SHOULD use the sent element in the bounce to determine that an error is from a forked message.</p>
-    <p>This rule is used to prevent some of the half-failure modes that have been an issue in other prototocols.</p>
```
Comment (horazont) [inline xep-0280.xml]: While we’re at it, can we fix that typo?

Comment (penguineer) [inline xep-0280.xml]: Hm.. funny I didn't see that typo when I mangled the sentence. 
Thanks for pointing it out, I'll add a correction this evening (UTC+1).

Comment (ge0rg) [inline xep-0280.xml]: @penguineer no need to hurry, I've added some more stuff on top of your commit and will make a PR for all of them based on yours. :)

**Comment by ge0rg** [comment]
@SamWhited @linuxwolf because I can't make a PR based on this PR until this one is merged, but I'd like to get the changes integrated in the same context, please review and include the following tree: https://github.com/ge0rg/xeps/commits/xep0280

It consists of:
* https://github.com/ge0rg/xeps/commit/36eb0de991621cbbba4634b9191f0e54453eecf6 removal of 'forking' (used to be #412)
* https://github.com/ge0rg/xeps/commit/16d37348d2c3a91fb870dd742533477d4abaa64e fix of the protocols typo noted by @horazont 
* https://github.com/ge0rg/xeps/commit/bd3018f6a27bd28637e03e8e0b62cf04a3652989 inversion of the "use 'sent' / 'received' element from error bounce" suggestion, because it's not guarnteed to be present.

**Comment by ge0rg** [comment]
Rationale for the third commit in https://tools.ietf.org/html/rfc6120#section-8.3.1 (emphasis mine; thx @horazont):

> 6. The entity that returns an error stanza MAY include the original
>    XML sent so that the sender can inspect and, if necessary,
>    correct the XML before attempting to resend (however, this is a
>    courtesy only and **the originating entity MUST NOT depend on
>    receiving the original payload)**.  Naturally, the entity MUST NOT
>    include the original data if it not well-formed XML, violates the
>    XML restrictions of XMPP (see under Section 11.1), or is
>    otherwise harmful (e.g., exceeds a size limit).

**Comment by penguineer** [comment]
I will add @ge0rg 's commits to this PR.

**Comment by penguineer** [comment]
Done adding, 

@horazont your issue is resolved.

**Comment by linuxwolf** [comment]
As an author, I am fine with these changes.

**Comment by ge0rg** [comment]
I won't make it to a PC any more today.  On top of this, I'd like to do some more changes to improve the MUC-PM handling, which I hoped to get some list feedback on. I suppose I'll just go on with them in the next days.

**Pull request merged**